src/Flow/Debounce.cs

```diff
@@ -41,19 +41,24 @@ public static async Task<bool> Debounce(string debouncerKey, TimeSpan timeToWait
                 return false;
             }
 
-            try
+            // credits: https://stackoverflow.com/a/30740088
+            var delayTask = Task.Delay(timeToWait, debouncerData.TokenSource.Token);
+            var continuationTask = delayTask.ContinueWith(task =>
             {
-                await Task.Delay(timeToWait, debouncerData.TokenSource.Token);
-                Cleanup();
+                // Don't throw the exception if the task cancellation was requested Transfer the insidie Inside delay.Task.IsCancelled.
+            });
 
-                return true;
-            }
-            catch (TaskCanceledException)
+            await continuationTask;
```

Comment:
(-1, complex behavior should be explained inline). Let's add this from the provided source.  

```suggestion
            // Awaits a continuation that does nothing. Since it does nothing and doesn't have a CancellationToken, it can't throw any exceptions
            await continuationTask;
```